outbound-dialer/src/main/java/org/motechproject/nms/outbounddialer/service/impl/TargetFileServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+import org.motechproject.nms.outbounddialer.web.contract.FileProcessedStatusRequest;
+import org.motechproject.scheduler.contract.RepeatingSchedulableJob;
+import org.motechproject.scheduler.service.MotechSchedulerService;
+import org.motechproject.server.config.SettingsFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.security.DigestOutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+@Service("targetFileService")
+public class TargetFileServiceImpl implements TargetFileService {
+    private static final String TARGET_FILE_TIME = "outbound-dialer.target_file_time";
+    private static final String MAX_QUERY_BLOCK = "outbound-dialer.max_query_block";
+    private static final String TARGET_FILE_MS_INTERVAL = "outbound-dialer.target_file_ms_interval";
+    private static final String TARGET_FILE_DIRECTORY = "outbound-dialer.target_file_directory";
+    private static final String TARGET_FILE_NOTIFICATION_URL = "outbound-dialer.target_file_notification_url";
+    private static final String GENERATE_TARGET_FILE_EVENT = "nms.obd.generate_target_file";
+
+    private SettingsFacade settingsFacade;
+    private MotechSchedulerService schedulerService;
+    private AlertService alertService;
+    private SubscriptionDataService subscriptionDataService;
+    private CallRetryDataService callRetryDataService;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TargetFileServiceImpl.class);
+
+
+    private void scheduleTargetFileGeneration() {
+        //Calculate today's fire time
+        DateTimeFormatter fmt = DateTimeFormat.forPattern("H:m");
+        String timeProp = settingsFacade.getProperty(TARGET_FILE_TIME);
+        DateTime time = fmt.parseDateTime(timeProp);
+        DateTime today = DateTime.now()
+                .withHourOfDay(time.getHourOfDay())
+                .withMinuteOfHour(time.getMinuteOfHour())
+                .withSecondOfMinute(0)
+                .withMillisOfSecond(0);
+
+        //Millisecond interval between events
+        String intervalProp = settingsFacade.getProperty(TARGET_FILE_MS_INTERVAL);
+        Long msInterval = Long.parseLong(intervalProp);
+
+        LOGGER.debug(String.format("The %s message will be sent every %sms starting %s",
+                GENERATE_TARGET_FILE_EVENT, msInterval.toString(), today.toString()));
+
+        //Schedule repeating job
+        MotechEvent event = new MotechEvent(GENERATE_TARGET_FILE_EVENT);
+        RepeatingSchedulableJob job = new RepeatingSchedulableJob(
+                event,          //MOTECH event
```

Comment:
Not fully sure about the inner workings of schedulable job, but just to confirm, this has no dependency on -

1. the time that the service is first deployed
 - so today vs. timeProp?
2. the running time of the event once its fired
- it doesnt keep sliding.